src/linear_algebra/matrix/block.lean

```diff
@@ -30,71 +30,47 @@ matrix, diagonal, det, block triangular
 
 -/
 
-open_locale big_operators
+open finset function order_dual
+open_locale big_operators matrix
 
 universes v
 
-variables {m n : Type*} [decidable_eq n] [fintype n] [decidable_eq m] [fintype m]
-variables {R : Type v} [comm_ring R]
+variables {α m n : Type*}
+variables {R : Type v} [comm_ring R] {M : matrix m m R} {b : m → α}
 
 namespace matrix
 
-lemma det_to_block (M : matrix m m R) (p : m → Prop) [decidable_pred p] :
-  M.det = (matrix.from_blocks (to_block M p p) (to_block M p (λ j, ¬p j))
-    (to_block M (λ j, ¬p j) p) (to_block M (λ j, ¬p j) (λ j, ¬p j))).det :=
-begin
-  rw ← matrix.det_reindex_self (equiv.sum_compl p).symm M,
-  rw [det_apply', det_apply'],
-  congr, ext σ, congr, ext,
-  generalize hy : σ x = y,
-  cases x; cases y;
-  simp only [matrix.reindex_apply, to_block_apply, equiv.symm_symm,
-    equiv.sum_compl_apply_inr, equiv.sum_compl_apply_inl,
-    from_blocks_apply₁₁, from_blocks_apply₁₂, from_blocks_apply₂₁, from_blocks_apply₂₂,
-    matrix.submatrix_apply],
-end
+section has_lt
+variables [has_lt α]
 
-lemma det_to_square_block (M : matrix m m R) {n : nat} (b : m → fin n) (k : fin n) :
-  (to_square_block M b k).det = (to_square_block_prop M (λ i, b i = k)).det :=
-by simp
+/-- Let `b` map rows and columns of a square matrix `M` to blocks indexed by `α`s. Then
+`block_triangular M n b` says the matrix is block triangular. -/
+def block_triangular (M : matrix m m R) (b : m → α) : Prop := ∀ ⦃i j⦄, b j < b i → M i j = 0
```

Comment:
I've split out #16155 to separate the renaming from the refactor, in the hope that it will make the diff clearer.